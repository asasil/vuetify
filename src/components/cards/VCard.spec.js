import { test } from '~util/testing'
import VCard from '~components/cards/VCard'

test('VCard.js', ({ mount, functionalContext }) => {
// describe('VCard.vue', () => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(VCard, functionalContext())

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a flat card', () => {
    const wrapper = mount(VCard, functionalContext({
      propsData: {
        flat: true
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a raised card', () => {
    const wrapper = mount(VCard, functionalContext({
      propsData: {
        raised: true
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a tile card', () => {
    const wrapper = mount(VCard, functionalContext({
      propsData: {
        tile: true
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a card with custom height', () => {
    const wrapper = mount(VCard, functionalContext({
      propsData: {
        height: '400px'
      }
    }))

    console.log(wrapper.hasStyle('height', '400px'))

    expect(wrapper.hasStyle('height', '400px')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a tile card', () => {
    const wrapper = mount(VCard, {
      propsData: {
        tile: true
      }
    })

    expect(wrapper.hasClass('card--tile')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
