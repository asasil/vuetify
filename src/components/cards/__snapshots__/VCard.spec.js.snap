// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCard.js should render a card with custom height 1`] = `

<div class="card"
     style="height: auto;"
>
</div>

`;

exports[`VCard.js should render a flat card 1`] = `

<div class="card"
     style="height: auto;"
>
</div>

`;

exports[`VCard.js should render a raised card 1`] = `

<div class="card"
     style="height: auto;"
>
</div>

`;

exports[`VCard.js should render a tile card 1`] = `

<div class="card"
     style="height: auto;"
>
</div>

`;

exports[`VCard.js should render a tile card 2`] = `

<!---->

`;

exports[`VCard.js should render component and match snapshot 1`] = `

<div class="card"
     style="height: auto;"
>
</div>

`;
